packages/perseus/src/widgets/python-program.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * This widget is for embedding Khan Academy Python programs.
+ */
+import {View} from "@khanacademy/wonder-blocks-core";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import {toAbsoluteUrl} from "../util/url-utils";
+
+import type {WidgetExports} from "../types";
+
+function getUrlFromProgramID(programID: any) {
+    const path = `/python-program/${programID}/embedded`;
+
+    return toAbsoluteUrl(path);
+}
+
+type Props = {
+    programID: string;
+    height: number;
+};
+
+type DefaultProps = {
+    height: Props["height"];
+};
+
+/* This renders the program in an iframe. */
+class PythonProgram extends React.Component<Props> {
+    static defaultProps: DefaultProps = {
+        height: 400,
+    };
+
+    render(): React.ReactNode {
+        const url = getUrlFromProgramID(this.props.programID);
+        const iframeStyle = {
+            height: this.props.height,
+            width: "100%",
+        } as const;
+
+        const sandboxOptions = [
+            "allow-popups",
+            "allow-same-origin",
+            "allow-scripts",
+            "allow-top-navigation",
+        ].join(" ");
+
+        // We sandbox the iframe so that we allowlist only the functionality
+        //  that we need. This makes it a bit safer in case some content
+        //  creator "went wild".
+        // http://www.html5rocks.com/en/tutorials/security/sandboxed-iframes/
+        return (
+            <View style={styles.container}>
+                <iframe
+                    sandbox={sandboxOptions}
+                    src={url}
+                    style={iframeStyle}
+                    allowFullScreen={true}
+                />
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    container: {
+        margin: "auto",
+        width: "100%",
+    },
+});
+
+export default {
+    name: "python-program",
+    displayName: "Python Program",
+    widget: PythonProgram,
+} as WidgetExports<typeof PythonProgram>;
```

Comment:
I just thought of this. We have no scoring implemented for this widget (ie. it can't be graded). So we should add `graded: false` to this structure to signal to Perseus that it shouldn't try (or expect) to grade instances of it. 

Ref: https://github.com/Khan/perseus/blob/main/packages/perseus/src/perseus-types.ts#L93

Reply:
Are you sure this is the right place for that? This has the type `WidgetExports`, which [doesn't have a graded property](https://github.com/Khan/perseus/blob/9f2eafe4c4649fd1ee1a83611b53908eb9cf3468/packages/perseus/src/types.ts#L409). It seems like it needs to be a property of the `Widget`, but I don't understand how objects of that type are used.

Reply:
Sorry Adam, I was digging through our types and got that wrong. 

Given that the initial target is to use this in Articles, let's leave this as is. If we do want to start using it as a helper in exercises, we'd have to do more work to provide scoring, but I think that's not required now that I've dug a bit further.